force-app/main/default/classes/FSL/WorkOrderImportController.cls

```diff
@@ -0,0 +1,456 @@
+            throw new AuraHandledException('Problem occured during creating Service Appointments');
+        }
+        return null;
+    }
+
+    private static List<WorkOrderImportModel.WorkOrderExcel> addWorkOrderPropertyIds (
+        List<WorkOrderImportModel.WorkOrderExcel> workOrderExcelList
+        ) {
+        Set <String> oldPropertyIdSet = new Set<String>();
+        for (WorkOrderImportModel.WorkOrderExcel w : workOrderExcelList) {
+            oldPropertyIdSet.add(w.PropertyId);
+        }
+        List<Property__c> properties = [SELECT Yardi_Property_ID__c
+                                        FROM Property__c
+                                        WHERE Yardi_Property_ID__c IN :oldPropertyIdSet];
+        Map<String, Id> propertyYardiIdMap = new Map<String, Id>();
+        for (Property__c p : properties) {
+            propertyYardiIdMap.put(p.Yardi_Property_ID__c, p.Id);
+        }
+        if (propertyYardiIdMap.keyset().size() < oldPropertyIdSet.size()) {
+            for (String pId : propertyYardiIdMap.keyset()) {
+                if (oldPropertyIdSet.contains(pId)) {
+                    oldPropertyIdSet.remove(pId);
+                }
+            }
+            for (WorkOrderImportModel.WorkOrderExcel w : workOrderExcelList) {
+                if (oldPropertyIdSet.contains(w.PropertyId)) {
+                    ErrorEntity e = new ErrorEntity('1',
+                                                    w.FakeId,
+                                                    new List<String>
+                                                    {'This PropertyId is missing in the DB'});
+                    errorsToFrontList.add(e);
+                }
+            }
+            return null;
+        }
+        for (WorkOrderImportModel.WorkOrderExcel w : workOrderExcelList) {
+            if (propertyYardiIdMap.keyset().contains(w.PropertyId)) {
+                w.NewPropertyId = propertyYardiIdMap.get(w.PropertyId);
+            }
+        }
+        return workOrderExcelList;
+    }
+
+    private static List<WorkOrderImportModel.WorkOrderExcel> addWorkOrderContactIds (
+        List<WorkOrderImportModel.WorkOrderExcel> workOrderExcelList
+        ) {
+        Set<String> excelEmails = new Set<String>();
+        Set<String> excelPhones = new Set<String>();
+        for (WorkOrderImportModel.WorkOrderExcel w : workOrderExcelList) {
+            excelEmails.add(w.CallerEmail);
+            excelPhones.add(w.CallerPhone);
+        }
+        List<Contact> contactsWithEmails = [SELECT Email
+                                            FROM Contact
+                                            WHERE Email IN :excelEmails];
+        List<Contact> contactsWithPhones = [SELECT Phone
+                                            FROM Contact
+                                            WHERE Phone IN :excelPhones];
+        Map<String, Id> emailsAndIdMap = new Map<String, Id>();
+        for (Contact c : contactsWithEmails) {
```

Comment:
Variable contactsWithEmails is not needed as it's used only in this for loop. Use SOQL for loop:
`for (Contact c : [SELECT Email FROM Contact WHERE Email IN :excelEmails]) {}`